### DISCUSSION TITLE
New File Options

### DISCUSSION THREAD

**Discussion by Natejoestev** [moudey/Shell#113] [General]
Before discovering Shell, i had to use the registry keys to add/remove file types to the New File menu.
Now that i use Shell, how can i change these options under New?
![image](https://user-images.githubusercontent.com/85044853/206828550-3949efb7-5947-4bcb-98b5-80b4d99919b3.png)

**Comment by moudey** [comment]
Hi @Natejoestev, Let's talk through examples with this [bulld](https://github.com/moudey/Shell/files/10180947/shell-test.zip)

Delete a shortcut item from New menu
```
static
{
    item(
         // Check when clicking in an empty space
         type='back'

         // The first parameter to search for the title of the item. 
         // The second is the location of the item
         find=['shortcut', '/new'] 

         // Item removal command
         vis=remove
     )
}
```

Add a custom item that creates a text file containing a message
```
dynamic
{
    item(title="my item in 'New Menu'" 
        parent="/new" 
        image=#0f0 
        cmd=@io.file.create('@(str.guid).txt', 'Hello World!')
    )
}
```

**Comment by Natejoestev** [comment]
does regedit still affect the menu, or is it all in the `shell.shl`?

**Comment by moudey** [comment]
Shell does not use the Windows registry to edit items.  Modifications are made through the configuration file.

**Comment by Natejoestev** [comment]
so how can i change the files in the `new` menu, add stuff like `open in Visual Studio`, and change the `file manage` options for the file context menu?

**Comment by moudey** [comment]
Please refer to the first comment

**Comment by Natejoestev** [comment]
so, everything that isn't in the `.shl` file now, is a default?

**Comment by Natejoestev** [comment]
what is wrong with line 53???
`line[53] column[8], Property unexpected "shell.shl"`:
![image](https://user-images.githubusercontent.com/85044853/210155391-d395ec4a-661d-4785-9e48-bed6f7c7ed40.png)

**Comment by moudey** [comment]
`type='back'` and `vis='remove'`

**Comment by moudey** [comment]
Please try this [build](https://github.com/moudey/Shell/files/10329755/shell-test.zip)
The settings file contains several ways to customize static and dynamic items, with the possibility of dividing the config file into several files to simplify work on it.

**Comment by s17534** [comment]
Hi! My issue relates to this discussion so I just chime in ;)

I can't disable New file menu (or edit it's contents), tried this but nothing happens as you can see below:

```
static
{
    item(
         // Check when clicking in an empty space
         type='back'

         // The first parameter to search for the title of the item. 
         // The second is the location of the item
         find=['shortcut', '/new'] 

         // Item removal command
         vis=remove
     )
}

```

![image](https://user-images.githubusercontent.com/61653293/226627925-9dd07c1e-b980-4b56-815a-f672c8d167e5.png)

also, adding new entry there isn't working, tried this sample:

```
dynamic
{
    item(title="my item in 'New Menu'" 
        parent="/new" 
        image=#0f0 
        cmd=@io.file.create('@(str.guid).txt', 'Hello World!')
    )
}

```

**Reply by moudey** [reply]
@s17534 Change the entries according to your language
```
static
{
    item(
         type='back'
         find=['skrót', '/nowy']  // Modify these values according to the display language
         vis="remove"
     )
}
```

```
dynamic
{
    item(title="my item in 'New Menu'" 
        parent="/nowy" // Modify these values according to the display language
        image=#0f0 
        cmd=@io.file.create('@(str.guid).txt', 'Hello World!')
    )
}
```

**Reply by s17534** [reply]
Thank you for fast response. I have used settings provided by you but unfortunately they also did not changed anything. In log I don't see anything that could point me to cause of said issue. Maybe I could somehow set debug log for app?

**Reply by moudey** [reply]
After changes press `ctrl` + `right click` to reload settings

**Reply by s17534** [reply]
Oh believe me, I tried that and restart explorer too. Even on english windows too:

![image](https://user-images.githubusercontent.com/61653293/226898196-f04ad207-2e0e-4b15-8b7f-f2b0c684d070.png)

**Reply by moudey** [reply]
Use the `in` property instead of [search, parent] and there is a bug when writing `/` at the beginning of the name
```
static
{
    item(
         type='back'
         find='shortcut' 
         in='new'
         vis="remove"
     )
}

dynamic
{
    item(title="my item in 'New Menu'" 
        parent="new"
        image=#0f0 
        cmd=@io.file.create('@(str.guid).txt', 'Hello World!')
    )
}

```

**Reply by s17534** [reply]
And now we are getting somewhere. By adding those lines:

`item(type='back' find='skrót' in='new' vis="remove")
item(type='back' find='Microsoft' in='new' vis="remove")
item(type='back' find='Folder' in='new' vis="remove")
item(type='back' find='Obraz' in='new' vis="remove")`

I got rid of every one of the items in the "Nowy" menu. Then I figured out why not just set one like this while commenting previous ones:

`
item(type='back' find='nowy' vis="remove")
`

Now I have no "Nowy" menu, Good Job! May I ask you, is it possible to move some of the items from other apps to my custom menu?

For example I have some "Git GUI here" or "Git Bash here" items and I want to move them to "Develop" menu. Is it possible?

**Reply by moudey** [reply]
It can be improved to remove multiple items with one command:
```
item(type='back' find='skrót|Microsoft|Folder|Obraz' in='new' vis="remove") 

// Or remove all items in New submenu
item(type='back' find='*' in='new' vis="remove") 

// Move items to submenu
item(find='Git GUI here|Git Bash here' menu='Develop')
```

**Reply by s17534** [reply]
It's perfect now, thank you :)
